src/dataplane-protocol/src/record.rs

```diff
@@ -41,147 +41,121 @@ pub trait AsyncBuffer {
+        let value = &self.0;
+        if matches!(inspect(value), ContentType::BINARY) {
+            write!(f, "values binary: ({} bytes)", self.len())
+        } else if value.len() < *MAX_STRING_DISPLAY {
+            write!(f, "{}", String::from_utf8_lossy(value))
+        } else {
+            write!(
+                f,
+                "{}...",
+                String::from_utf8_lossy(&value[0..*MAX_STRING_DISPLAY])
+            )
         }
     }
 }
 
 impl Display for DefaultAsyncBuffer {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self.0 {
-            Some(ref value) => {
-                if matches!(inspect(value), ContentType::BINARY) {
-                    write!(f, "binary: ({} bytes)", self.len())
-                } else if value.len() < *MAX_STRING_DISPLAY {
-                    write!(f, "{}", String::from_utf8_lossy(value))
-                } else {
-                    write!(
-                        f,
-                        "{}...",
-                        String::from_utf8_lossy(&value[0..*MAX_STRING_DISPLAY])
-                    )
-                }
-            }
-            None => write!(f, ""),
+        let value = &self.0;
+        if matches!(inspect(value), ContentType::BINARY) {
+            write!(f, "binary: ({} bytes)", self.len())
+        } else if value.len() < *MAX_STRING_DISPLAY {
+            write!(f, "{}", String::from_utf8_lossy(value))
+        } else {
+            write!(
+                f,
+                "{}...",
+                String::from_utf8_lossy(&value[0..*MAX_STRING_DISPLAY])
+            )
         }
     }
 }
 
 impl From<String> for DefaultAsyncBuffer {
     fn from(value: String) -> Self {
-        Self(Some(value.into_bytes()))
+        Self(value.into_bytes())
     }
 }
 
 impl From<Vec<u8>> for DefaultAsyncBuffer {
     fn from(value: Vec<u8>) -> Self {
-        Self(Some(value))
+        Self(value)
     }
 }
 
 impl<'a> From<&'a [u8]> for DefaultAsyncBuffer {
     fn from(bytes: &'a [u8]) -> Self {
         let value = bytes.to_owned();
-        Self(Some(value))
+        Self(value)
     }
 }
 
 impl Encoder for DefaultAsyncBuffer {
-    fn write_size(&self, _version: Version) -> usize {
-        self.0.var_write_size()
+    fn write_size(&self, version: Version) -> usize {
+        self.0.write_size(version)
     }
 
-    fn encode<T>(&self, src: &mut T, _version: Version) -> Result<(), Error>
+    fn encode<T>(&self, src: &mut T, version: Version) -> Result<(), Error>
     where
         T: BufMut,
     {
-        self.0.encode_varint(src)?;
-
-        Ok(())
+        self.0.encode(src, version)
     }
 }
 
 impl Decoder for DefaultAsyncBuffer {
-    fn decode<T>(&mut self, src: &mut T, _version: Version) -> Result<(), Error>
+    fn decode<T>(&mut self, src: &mut T, version: Version) -> Result<(), Error>
     where
         T: Buf,
     {
         trace!("decoding default asyncbuffer");
-        self.0.decode_varint(src)?;
```

Comment:
should keep variant since it reduce size

Reply:
That's what https://github.com/infinyon/fluvio-protocol/pull/4 does but it seems that `Vec<u8>` shouldn't implement `EncodeVarInt`?

Reply:
Let's merge https://github.com/infinyon/fluvio-protocol/pull/5 and publish it to crates.io before merging this.